universal-application-tool-0.0.1/app/repository/QuestionRepository.java

```diff
@@ -33,6 +33,52 @@ public QuestionRepository(EbeanConfig ebeanConfig, DatabaseExecutionContext exec
         executionContext);
   }
 
+  static class PathConflictDetector {
+    private boolean conflict = false;
+    private final String newPath;
+
+    PathConflictDetector(String newPath) {
+      this.newPath = checkNotNull(newPath);
+    }
+
+    public Boolean hasConflict() {
+      return Boolean.valueOf(conflict);
+    }
+
+    public boolean checkConflict(Question question) {
+      boolean proceed = true;
+      if (pathConflicts(question.getPath(), newPath)) {
+        conflict = true;
+        proceed = false;
+      }
+      return proceed;
+    }
+
+    public static boolean pathConflicts(String path, String otherPath) {
+      path = path.toLowerCase();
+      otherPath = otherPath.toLowerCase();
+      return path.startsWith(otherPath) || otherPath.startsWith(path);
+    }
+  }
+
+  public CompletionStage<Boolean> lookupPathConflict(String newPath) {
```

Comment:
Suggestion (blocking): this method is called `lookupPathConflict` but returns a `boolean`. It seems likely that the question it's in conflict with could be useful for displaying a helpful error message so I recommend returning the question it's in conflict with and changing the name to `findConflictingQuestion`

Reply:
Done.